atsign-foundation/at_server
# fix: refactor error messages and JSON encode the error responses.

**- What I did**
- Return error codes and error description on exception.
- Generate the exception type from the error codes on the exception.

**- How I did it**
- When exception is raised, JSON encode the error codes and error descriptions.
- On receiving the exception, decode the JSON, and create the exception type from the error code and set the description.

**- How to verify it**
- Run the scenarios that trigger exception - look for a non existent key and verify the `KeyNotFoundException` type is returned along with error code and error description.

**- Description for the changelog** 
- Forward the error codes and error description.



**-Tests from at_server command line**

Look for a key when root server is not started.
@ sitaram@lookup:phone.wavi@murali
error:{"errorCode":"AT0011","errorDescription":"Internal server exception : Failed connecting to root server url vip.ve.atsign.zone on port 64"}

Look for a key when target secondary is not started.
@ sitaram@lookup:phone.wavi@murali
error:{"errorCode":"AT0007","errorDescription":"Secondary server not found : unable to connect to secondary"}

Look for a non existent key
@ sitaram@lookup:phone.wavi@murali
error:{"errorCode":"AT0015","errorDescription":"key not found : Exception: @ sitaram:phone.wavi@ murali does not exist in keystore"}
